client/my-sites/email/email-non-domain-owner-message/index.tsx

```diff
@@ -0,0 +1,140 @@
+import { SiteDetails } from '@automattic/data-stores';
+import { TranslateResult, useTranslate } from 'i18n-calypso';
+import { stringify } from 'qs';
+import { useDispatch } from 'react-redux';
+import { useDomainOwnerUserName } from 'calypso/components/data/query-domain-owner-username';
+import PromoCard from 'calypso/components/promo-section/promo-card';
+import { ResponseDomain } from 'calypso/lib/domains/types';
+import { emailManagementPurchaseNewEmailAccount } from 'calypso/my-sites/email/paths';
+import { recordTracksEvent } from 'calypso/state/analytics/actions';
+
+import './style.scss';
+
+type EmailNonDomainOwnerMessageProps = {
+	domain?: ResponseDomain;
+	selectedSite?: SiteDetails | null;
+	source: 'email-comparison' | 'email-management';
+	usePromoCard: boolean;
+};
+
+export const EmailNonDomainOwnerMessage = ( props: EmailNonDomainOwnerMessageProps ) => {
+	const { domain, selectedSite, source, usePromoCard } = props;
+
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+
+	const ownerUserName = useDomainOwnerUserName( selectedSite, domain );
+
+	const isPrivacyAvailable = domain?.privacyAvailable;
+
+	const buildQueryString = ( parameters = {} ) =>
+		parameters ? stringify( parameters, { addQueryPrefix: true, skipNulls: true } ) : '';
```

Comment:
We could move this function definition outside `EmailNonDomainOwnerMessage`, no?